libs/fabric/src/lib/components/core/shared/changeable-items.directive.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AfterContentInit, ContentChildren, EventEmitter, OnDestroy, Output, QueryList } from '@angular/core';
+
+import { ItemChangedPayload } from '../../core/declarative/item-changed.payload';
+import { ChangeableItemsHelper, IChangeableItemsContainer } from './changeable-helper';
+import { ChangeableItemDirective } from './changeable-item.directive';
+
+/**
+ * Parent class for wrapper directive for multiple ChangeableItemDirectives
+ */
+export abstract class ChangeableItemsDirective<TItem>
+  implements AfterContentInit, IChangeableItemsContainer<TItem>, OnDestroy {
+  @ContentChildren(ChangeableItemDirective)
+  readonly directiveItems: QueryList<ChangeableItemDirective<TItem>>;
+
+  @Output()
+  get onChildItemChanged(): EventEmitter<ItemChangedPayload<string, TItem>> {
+    return this.changeableItemsHelper && this.changeableItemsHelper.onChildItemChanged;
+  }
+  @Output()
+  get onItemsChanged(): EventEmitter<QueryList<ChangeableItemDirective<TItem>>> {
+    return this.changeableItemsHelper && this.changeableItemsHelper.onItemsChanged;
+  }
+
+  private changeableItemsHelper: ChangeableItemsHelper<TItem>;
+
+  abstract get items(): TItem[];
+
+  ngAfterContentInit() {
+    this.changeableItemsHelper = new ChangeableItemsHelper(this.directiveItems);
+    this.changeableItemsHelper.afterContentInit();
```

Comment:
These two lines one after the other make me think that there's no need for `afterContentInit()` in `ChangeableItemsHelper`, since it can only be created in `ngAfterContentInit()` anyway.

WDYT?

Reply:
I definitely agree. I think I only realized ChangeableItemsHelper had to be initialized in `ngAfterContentInit` after I already created `afterContentInit` for it. The `afterContentInit` logic makes more sense living in the constructor instead.